yesod-test/Yesod/Test.hs

```diff
@@ -754,6 +757,61 @@ htmlNoneContain query search = do
     found -> failure $ "Found " <> T.pack (show $ length found) <>
                 " instances of " <> T.pack search <> " in " <> query <> " elements"
 
+-- | Queries the HTML using a CSS selector, and all matched elements must contain
+-- the given string verbatim.
+--
+-- ==== __Examples__
+--
+-- > {-# LANGUAGE OverloadedStrings #-}
+-- > get HomeR
+-- > htmlAllContainPreEscaped "meta" " "content=\"site description\""-- Every <meta> tag contain the string "content=site description" verbatim
+--
+-- Since 16.6.6
+htmlAllContainPreEscaped :: HasCallStack => Query -> String -> YesodExample site ()
+htmlAllContainPreEscaped query search = do
+  matches <- htmlQuery query
+  case matches of
+    [] -> failure $ "Nothing matched css query: " <> query
+    _ -> liftIO $ HUnit.assertBool ("Not all "++T.unpack query++" contain "++search  ++ " matches: " ++ show matches) $
+          DL.all (DL.isInfixOf search) (map (TL.unpack . decodeUtf8) matches)
```

Comment:
Except for the escaping, this is the exact same code as `htmlAllContain`, it might make refactor this a bit.